crates/chia-consensus/src/test_generators.rs

```diff
@@ -353,59 +373,63 @@ fn run_generator(#[case] name: &str) {
         }
 
         // now lets check get_coinspends_for_trusted_block
-        let vec_of_slices: Vec<&[u8]> = block_refs.iter().map(std::vec::Vec::as_slice).collect();
+        // but only if we trust it not to do shenanigans
+        if run_generator_one {
+            let vec_of_slices: Vec<&[u8]> =
+                block_refs.iter().map(std::vec::Vec::as_slice).collect();
 
-        let result = get_coinspends_for_trusted_block(
-            &TEST_CONSTANTS,
-            &Program::new(generator.clone().into()),
-            &vec_of_slices,
-            flags,
-        );
-
-        let result2 = get_coinspends_with_conditions_for_trusted_block(
-            &TEST_CONSTANTS,
-            &Program::new(generator.clone().into()),
-            &vec_of_slices,
-            flags,
-        );
-
-        if let Ok(ref conds) = conds2 {
-            // if run_block_generator2 is OK then check we're equal
-            let coinspends = result.expect("get_coinspends");
-            let coinspends2 = result2.expect("get_coinspends_with_conds");
-            for (i, spend) in conds.spends.iter().enumerate() {
-                let runnable = {
-                    let mut a = make_allocator(flags);
-                    coinspends[i]
-                        .puzzle_reveal
-                        .run(
-                            &mut a,
-                            flags,
-                            TEST_CONSTANTS.max_block_cost_clvm,
-                            &coinspends[i].solution,
-                        )
-                        .is_ok()
-                };
-                assert!(runnable);
-                let parent_id = a2.atom(spend.parent_id);
-                assert_eq!(
-                    parent_id.as_ref(),
-                    coinspends[i].coin.parent_coin_info.as_slice()
-                );
-                let puzhash = a2.atom(spend.puzzle_hash);
-                assert_eq!(puzhash.as_ref(), coinspends[i].coin.puzzle_hash.as_slice());
-                assert_eq!(spend.coin_amount, coinspends[i].coin.amount);
+            let result = get_coinspends_for_trusted_block(
+                &TEST_CONSTANTS,
+                &Program::new(generator.clone().into()),
+                &vec_of_slices,
+                flags,
+            );
 
+            if let Ok(ref conds) = conds2 {
+                // if run_block_generator2 is OK then check we're equal
+                let coinspends = result.expect("get_coinspends");
                 // check that we're getting the same info from get_coinspends_with_conditions_for_trusted_block
-                assert_eq!(
-                    parent_id.as_ref(),
-                    coinspends2[i].0.coin.parent_coin_info.as_slice()
-                );
-                assert_eq!(
-                    puzhash.as_ref(),
-                    coinspends2[i].0.coin.puzzle_hash.as_slice()
+                let result2 = get_coinspends_with_conditions_for_trusted_block(
```

Comment:
Was there a good reason to move this inside the if-block, after checking that `conds2` succeeded? I suppose we don't use the resulting value outside the if-block